queue/group_util.go

```diff
@@ -187,6 +188,18 @@ func (c *cacheImpl) Prune(ctx context.Context) error {
 							defer recovery.LogStackTraceAndContinue("panic in queue waiting")
 							defer close(wait)
 
+							// It's possible for a job to be enqueued in between
+							// the queue completeness check and closing the
+							// queue. In this case, job that was just enqueued
+							// will not execute unless the queue is restarted.
+							// This log makes it easier to identify if/when this
+							// occurs.
+							grip.Info(message.Fields{
+								"message":  "pruning completed queue",
```

Comment:
I assume you're talking about [L188](https://github.com/mongodb/amboy/pull/195/files#diff-a11c2e053c1c36554e5dd94c390929edc292e5ae6aba85e0966628263d93f64eR188)? I think that's just a safety mechanism to avoid panicking the entire application. This log just makes it easier to trace when a queue group gets pruned and correlate it with a stranded job, if it ever happens.

I think it could be possible to fix, but I don't think it's that worthwhile, since it would require `Put()` to stop accepting jobs when it's not actively running. This log is mostly for edge case handling.